app-emacs/treemacs/treemacs-2.9.ebuild

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+NEED_EMACS="26.1"
+
+inherit elisp
+
+DESCRIPTION="A tree layout file explorer for Emacs"
+HOMEPAGE="https://github.com/Alexander-Miller/treemacs"
+SRC_URI="https://github.com/Alexander-Miller/treemacs/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="evil magit icons-dired" # some flags need packages not yet in gentoo
+
+RDEPEND=">=app-emacs/ace-window-0.10.0_p20200706
+	>=app-emacs/ht-2.3
+	>=app-emacs/hydra-0.15.0
+	>=app-emacs/pfuture-1.9
+	>=app-emacs/cfrs-1.3.2
+	evil? ( app-emacs/evil )
+	magit? ( app-emacs/magit )"
+DEPEND="${RDEPEND}"
+
+RESTRICT="test" # test requires cask and buttercup.el which are not packaged
+SITEFILE="50treemacs-gentoo.el"
+DOCS=( README.org Changelog.org Extensions.org )
+
+src_unpack() {
+	elisp_src_unpack
+	mv "${S}"/src/{elisp,scripts}/* "${S}/" || die
+	if use evil; then
+		mv "${S}/src/extra/treemacs-evil.el" "${S}/" || die
+	fi
+	if use magit; then
+		mv "${S}/src/extra/treemacs-magit.el" "${S}/" || die
+	fi
+	if use icons-dired; then
+		mv "${S}/src/extra/treemacs-icons-dired.el" "${S}/" || die
+	fi
+}
+
+src_compile() {
+	elisp-make-autoload-file "${S}/${PN}-autoload.el" "${S}/"
+	elisp_src_compile
+}
+
+src_install() {
+	elisp_src_install
+	mkdir -p "${D}/usr/share/${PN}/icons/" || die
+	cp -r "${S}/icons/default/" "${D}/usr/share/${PN}/icons/" || die
```

Comment:
What are these icons for?

Do we need to use [doicon](https://devmanual.gentoo.org/eclass-reference/desktop.eclass/index.html) from desktop.eclass?

Reply:
The icons are used in the treemacs buffer. It shows various different icons depending on what type of file is at a certain path. For example it shows a folder icon for directories, a bandaid for patch files, a sheet of paper for plain text files, etc.

From what I can tell doicon is used for installing icons used for application menus that use desktop files right? I certainly agree that there may be a better way to do this and was very unsure of the original way when I wrote the ebuild! 

Also the symlink is needed because treemacs expects the icons to be in the directory where the elisp sources are. I don't know if it would be okay to just put them in ${SITELISP}/treemacs/icons/ or not, but that may not be terrible. 

Let me know what you think!

Reply:
Ohh, duh, I see. Yeah, I think you're right here.

I like the /usr/share method better too with the symlink.